src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHashShared.cs

```diff
@@ -0,0 +1,785 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+#if NET7_0_OR_GREATER
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Runtime.Intrinsics.X86;
+#endif
+
+namespace System.IO.Hashing
+{
+    /// <summary>Shared implementation of the XXH3 hash algorithm for 64-bit in <see cref="XxHash3"/> and <see cref="XxHash128"/> version.</summary>
+#if NET5_0_OR_GREATER
+    [SkipLocalsInit]
+#endif
+    internal static unsafe class XxHashShared
+    {
+        /// <summary>XXH3 produces 8-byte hashes.</summary>
```

Comment:
This is strange to have in the shared file.

Reply:
Fixed by commit b0fe87695a858d5a8ca02b19d17eb18f2e486884